### DISCUSSION TITLE
Problem with watch() and useWatch()

### DISCUSSION THREAD

**Discussion by juliopavila** [react-hook-form/react-hook-form#3250] [Q&A] [answered]
I have a question when I use an unregister to delete a field from JSON Object the `useWatch` hook doesn’t update but if I use `watch()` function that’s work the problem that I have using ` watch()` is they make a lot of renders when I send it to father component

**Comment by bluebill1049** [comment, marked as answer]
wait this should already be fixed, which version are you on?

**Reply by juliopavila** [reply]
@bluebill1049 

This `"react-hook-form": "^6.9.5",` to fix it I made this

```
 useEffect(() => {
    let currentData = watch();
    if (!_.isEqual(currentData, currentValues)) {
      props.valuesCompleted(currentData);
      setCurrentValues(currentData);
    }
  }, [watch()]);
```
Create a state, and compare the state against the returned value of `watch()` to avoid bucle

**Reply by bluebill1049** [reply]
curious why do you need to do the comparison? can't you do the logic in the render base on what's watch returns?

**Reply by juliopavila** [reply]
@bluebill1049 

let me explain, I made a form generator that component receive a JSON object and render all field well, I this form generator could hide and show some fields depends on validations for that reason I used `useWatch()` hook for pass current data to father component for making the validations, and my error was when I register or unregister element after the validations the `useWatch()` hook doesn’t update for that reason I applied for work with the only `watch()` and that validation is for avoiding re-renders
